src/NzbDrone.Core/Download/TrackedDownloads/DownloadMonitoringService.cs

```diff
@@ -113,12 +121,48 @@ private List<TrackedDownload> ProcessClientDownloads(IDownloadClient downloadCli
 
         private void RemoveCompletedDownloads(List<TrackedDownload> trackedDownloads)
         {
-            foreach (var trackedDownload in trackedDownloads.Where(c => c.DownloadItem.CanBeRemoved && c.State == TrackedDownloadStage.Imported))
+            var removableDownloads = trackedDownloads.Where(c =>
+            {
+                if (c.Protocol != DownloadProtocol.Torrent)
+                {
+                    return c.DownloadItem.CanBeRemoved && c.State == TrackedDownloadStage.Imported;
+                }
+
+                return TorrentHasFinishedSeeding(c);
+            });
+
+            foreach (var trackedDownload in removableDownloads)
             {
                 _eventAggregator.PublishEvent(new DownloadCompletedEvent(trackedDownload));
             }
         }
 
+        private bool TorrentHasFinishedSeeding(TrackedDownload download)
+        {
+            var historyItem = _historyService.Find(download.DownloadItem.DownloadId, HistoryEventType.Grabbed).Last();
```

Comment:
@markus101 I'm not sure about this line, is it okay like this?

Reply:
Sorry! Missed this comment. Not sure what else we could do here instead, the last grabbed with that ID should be the one we're concerned about. Only possible issue would be torrents with the same hash from multiple trackers, but not sure we can do anything about that.